src/components/Error/ErrorBoundary.tsx

```diff
@@ -28,19 +32,52 @@ export default class ErrorBoundary extends Component<PropsWithChildren<ErrorBoun
     this.props.onError?.(error, errorInfo)
   }
 
+  getErrorDialogHeader(error: Error): string {
+    if (error instanceof MetaMaskConnectionError) {
```

Comment:
These messages should be encoded into the errors, so it doesn't need to be duplicated. eg the Error that you extend from in src/errors.ts should have certain properties used to fill out the dialog, like

    abstract class WidgetError extends Error {
      header: string
      action: string
    }